### PR TITLE
Change language generation inputs

### PR DESCRIPTION
**Pull request by ahmedlhanafy**
Fixes: https://lgtools.visualstudio.com/LG%20For%20Bots/_workitems/edit/76/
## Description
As per the LG team discussion, this PR addresses the addition of the version and locale to the language generation inputs and remove the default values. 

## Specific Changes
  - Specify language generation version and locale  
  - Provide the option to override the default runtime and issue token endpoints 
  - Remove all default inputs and require them upfront and add validation steps
  - Update tests and samples to reflect the new change

### CODE REVIEW

**Comment by ahmedlhanafy** [comment]
cc @vishwacsena

**Comment by MahEzzat** [comment]
Add me as a reviewer

File: libraries/botbuilder-ai/src/languageGenerationResolver.ts [hunk 1]
```diff
@@ -40,17 +40,26 @@ export interface LanguageGenerationApplication {
...
 
-	/* Congnetive service authorization key */
-	endpointKey: string;
+	/** Language generation region */
+	applicationRegion: string;
+
+	/** Language generation version */
+	applicationVersion: string;
+
+	/* Congnetive service subscription key */
```
Comment (MahEzzat) [inline libraries/botbuilder-ai/src/languageGenerationResolver.ts]: nit: Cognitive services

File: libraries/botbuilder-ai/tests/languageGenerationResolver.test.js [hunk 2]
```diff
@@ -915,13 +918,13 @@ describe('LanguageGenerationResolver', () => {
 			}
 		});
 
-		it('should throw an error if an invalid key or application id are given', async () => {
+		it('should throw an error if an invalid options object is provided', async () => {
 			let err = null;
 			try {
 				const resolver = new LanguageGenerationResolver(
 					{
 						applicationId: '',
```
Comment (MahEzzat) [inline libraries/botbuilder-ai/tests/languageGenerationResolver.test.js]: Shouldn't you fill the rest of the LG app object parameters?

**Comment by ahmedlhanafy** [comment]
@cahann Merge this one after merging #707

**Pull request merged**